### PR TITLE
improved the code using code climate

### PR DESCRIPTION
**Pull request by nendhruv**
used code climate to solve the issues and improved the code. Related to #84

### CODE REVIEW

File: javascripts/content_scripts/privly.js [hunk 1]
```diff
@@ -763,8 +757,6 @@ var privly = {
 
     "use strict";
 
-    return;
```
Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: This is a semantic change. I know CodeClimate will complain about it, but it should not be included in this PR.

**Comment by nendhruv** [comment]
@smcgregor made it as it was before

File: javascripts/content_scripts/privly.js [hunk 2]
```diff
@@ -359,42 +359,36 @@ var privly = {
...
-    iFrame.setAttribute("marginwidth","0");
-    iFrame.setAttribute("marginheight","0");
-    iFrame.setAttribute("height","1px");
-    iFrame.setAttribute("style","width: 100%; height: 32px; " +
-      "overflow: hidden;");
-    iFrame.setAttribute("scrolling","no");
-    iFrame.setAttribute("overflow","hidden");
+     for(var key in attrs) {
+         iFrame.setAttribute(key, attrs[key]);
+       }
```
Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: Closing bracket should have the same indentation as the block that opened it.

File: javascripts/content_scripts/privly.js [hunk 3]
```diff
@@ -299,8 +299,8 @@ var privly = {
       var notUpdated = [];
       elements.forEach(
         function(a){
-          if ( ! a.textContent.indexOf("privlyInject1") > 0 // Optimization
-            || ! privly.correctIndirection.testAndCopyOver(a, a.textContent) ) {
+          if (  a.textContent.indexOf("privlyInject1") < 0 || ! privly.correctIndirection.testAndCopyOver(a, a.textContent) ) {
```
Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: The optimization comment should go above the "if" statement.

File: javascripts/content_scripts/privly.js [hunk 4]
```diff
@@ -144,13 +144,13 @@ var privly = {
   /**
    * Make plain text links into anchor elements.
    */
+
```
Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: accidentally changed whitespace.

**Comment by smcgregor** [comment]
A few more small comments. If changed, this is ready to test then merge.

**Comment by nendhruv** [comment]
@smcgregor done.

File: javascripts/content_scripts/privly.js [hunk 5]
```diff
@@ -299,8 +298,8 @@ var privly = {
       var notUpdated = [];
       elements.forEach(
         function(a){
-          if ( ! a.textContent.indexOf("privlyInject1") > 0 // Optimization
-            || ! privly.correctIndirection.testAndCopyOver(a, a.textContent) ) {
+          // Optimization
+          if (  a.textContent.indexOf("privlyInject1") < 0 || ! privly.correctIndirection.testAndCopyOver(a, a.textContent) ) {
```
Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: This should be `<=` to maintain the same semantics. Also, it is preferable to have this on two lines so it doesn't break the width limitations.

**Comment by smcgregor** [comment]
Just one more change.

File: javascripts/content_scripts/privly.js (hunk 5 above)
Comment (nendhruv) [inline javascripts/content_scripts/privly.js]: Code Climate complains if this is on two lines

Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: I thought that might be the case. It should also complain if the line is too long, but that would likely have too many false positives.

Comment (nendhruv) [inline javascripts/content_scripts/privly.js]: Well in this case it did not. So should I break it into two lines only?

Comment (smcgregor) [inline javascripts/content_scripts/privly.js]: I think the best thing to do here is to move each of the boolean statements into their own local variable. Something roughly like:

`var taggedForInjection = ...`  
`var copiedOver = taggedForInjection || ! privly.correctIndirection.testAndCopyOver(a, a.textContent)`

then the if statement.

I would also be OK with dropping this change and letting someone concentrate on this in the future for their "level 0." You have other changes here.

Comment (nendhruv) [inline javascripts/content_scripts/privly.js]: For now lets drop it for someone else.

**Comment by smcgregor** [comment]
If you don't comment about the changes you made to the PR, it won't make its way back into the review queue.

**Comment by nendhruv** [comment]
@smcgregor I'll make sure I comment from the next time. The latest commit has changes to maintain the semantics you mentioned above.

**Comment by smcgregor** [comment]
The code here looks good to go. I'll test it to make sure then merge.

**Comment by smcgregor** [comment]
I had to squash the commits, this is merged:

https://github.com/privly/privly-chrome/commit/8f98450a651e48a597a0405d7ad5b0572bb814a6
